pkg/provider/vault/vault.go

```diff
@@ -1193,6 +1208,18 @@ func setLdapAuthToken(ctx context.Context, v *client) (bool, error) {
 	return false, nil
 }
 
+func setUserPassAuthToken(ctx context.Context, v *client) (bool, error) {
+	userPassAuth := v.store.Auth.UserPass
+	if userPassAuth != nil {
+		err := v.requestTokenWithUserPassAuth(ctx, userPassAuth)
+		if err != nil {
+			return true, err
```

Comment:
Why return true here, is there a need for it? From what i can see in `requestTokenWithUserPassAuth` it is either an error or nil if successful.

Reply:
well good point, i only tried to to be constant with the other functions which are all doing it like this ;)
it is used to signal that a token exists:
        tokenExists, err = setUserPassAuthToken(ctx, v)

but yeah if i change it we have to change the whole function: https://github.com/external-secrets/external-secrets/blob/80d48c44e1300a4e86e57886d95abfb44f45875e/pkg/provider/vault/vault.go#L1101

Reply:
I just meant that the return `true` is odd, when you receive an error. From what i can see in the `requestTokenWithUserPassAuth` there seems to be no case where you return an error AND a token was created.

Hence we can simply `return false, err` here.
But you're right, that way it's consistent with the other funcs, which all should be changed :sweat_smile: 

We'll leave that for a different PR :)